nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -0,0 +1,316 @@
+            .description("The number of minutes (up to 5 minutes) of streaming data to be requested after a " +
+                    "disconnect. Only available for project with academic research access. See " +
+                    "https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/integrate/" +
+                    "recovery-and-redundancy-features")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("All tweets will be routed to this relationship.")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+    private Set<Relationship> relationships;
+
+    private TwitterStreamAPI api;
+
+    private volatile BlockingQueue<String> messageQueue;
+
+    private void emptyQueue() {
+        while (!messageQueue.isEmpty()) {
+            messageQueue.poll();
+        }
+    }
+
+    @Override
+    protected void init(ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(ENDPOINT);
+        descriptors.add(BASE_PATH);
+        descriptors.add(BEARER_TOKEN);
+        descriptors.add(QUEUE_SIZE);
+        descriptors.add(BATCH_SIZE);
+        descriptors.add(TWEET_FIELDS);
+        descriptors.add(USER_FIELDS);
+        descriptors.add(MEDIA_FIELDS);
+        descriptors.add(POLL_FIELDS);
+        descriptors.add(PLACE_FIELDS);
+        descriptors.add(EXPANSIONS);
+        descriptors.add(BACKFILL_MINUTES);
+
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return this.descriptors;
+    }
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        messageQueue = new LinkedBlockingQueue<>(context.getProperty(QUEUE_SIZE).asInteger());
+
+        api = new TwitterStreamAPI(context, messageQueue, getLogger());
+        api.setBasePath(context.getProperty(BASE_PATH).getValue());
+        final String endpointName = context.getProperty(ENDPOINT).getValue();
+        if (ENDPOINT_SAMPLE.getValue().equals(endpointName)) {
+            api.start(TwitterStreamAPI.SAMPLE_ENDPOINT);
+        } else if (ENDPOINT_SEARCH.getValue().equals(endpointName)) {
+            api.start(TwitterStreamAPI.SEARCH_ENDPOINT);
+        } else {
+            throw new AssertionError("Endpoint was invalid value: " + endpointName);
```

Comment:
Use `ProcessException`

Reply:
Will change all the assertion errors into ProcessExceptions, thanks for the comment!